### PR TITLE
(docs): update api docs for Agent, Transaction and Span

### LINKED ISSUE #126: Document setting tags on individual transactions and spans

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ Create separate API docs for Agent, Transaction and Span which makes things cleaner and easier for the users of the agent. 
+ fixes elastic/apm-agent-rum-js#126 #219

### CODE REVIEW

File: docs/api.asciidoc [hunk 1]
```diff
@@ -1,275 +1,21 @@
...
 == API Reference
 
+The API reference documentation is divided into three parts:
 
-[float]
-[[agent-api]]
-=== Agent API
-You can access agent API after initializing the agent: 
+* <<agent-api,The `Agent` API>> - All functions and properties on the `Agent` object.
+An instance of the `Agent` object is acquired by calling the `init method` the agent either via script element on the page or require the `elastic-apm-rum` module in Node.js
```
Comment (bmorelli25) [inline docs/api.asciidoc]:
```suggestion
An instance of the `Agent` object is acquired by calling the `init method` the agent either via script element on the page or require the `elastic-apm-rum` module in Node.js.
```

File: docs/api.asciidoc [hunk 2]
```diff
@@ -1,275 +1,21 @@
...
+The `Agent` instance is usually referred to by the variable `apm` in this documentation
```
Comment (bmorelli25) [inline docs/api.asciidoc]:
```suggestion
The `Agent` instance is usually referred to by the variable `apm` in this documentation.
```

File: docs/api.asciidoc [hunk 3]
```diff
@@ -1,275 +1,21 @@
...
 
-[source,js]
-----
-var apm = require('@elastic/apm-rum').init(...)
-----
+* <<transaction-api,The `Transaction` API>> - All functions and properties on the `Transaction` object.
+An instance of the `Transaction` object is acquired by calling `apm.startTransaction()`
```
Comment (bmorelli25) [inline docs/api.asciidoc]:
```suggestion
An instance of the `Transaction` object is acquired by calling `apm.startTransaction()`.
```

File: docs/api.asciidoc [hunk 4]
```diff
@@ -1,275 +1,21 @@
...
 
+* <<span-api,The `Span` API>> - All functions and properties on the `Span` object.
+An instance of the `Span` object is acquired by calling `apm.startSpan()`
```
Comment (bmorelli25) [inline docs/api.asciidoc]:
```suggestion
An instance of the `Span` object is acquired by calling `apm.startSpan()`.
```

**Comment by vigneshshanmugam** [comment]
Thanks @bmorelli25 Applied your suggestions :)

**Pull request merged**